### PR TITLE
docs - upgrade bemetalsmith to 5.1.x

### PR DESCRIPTION
**Pull request by ben-polinsky**
BeMetalsmith 5.1 includes:
  - improved sidebar navigation styles and leftNav.md implementation
  - deprecated types filtered from search results by default
  - fix for internal accessors not being displayed
  - fix for search not working from sandbox page
  - fs-extra removal in generation script
  - removal of getting started section - combined with main learning document.
  - removal of unused code and assets from 1.x-2.x docs

### CODE REVIEW

File: docs/bis/leftNav.md [hunk 1]
```diff
@@ -1,87 +1,131 @@
...
+  - [BIS Schema KindOfQuantities](./guide/other-topics/kindOfQuantities.md)
+  - [BIS Schema Validation](./guide/other-topics/bis-schema-validation.md)
+
+- #### References
+
+  - [BIS Glossary](./guide/references/glossary.md)
+  - [Class-diagram Conventions](./guide/references/class-diagram-conventions.md)
+  - [Instance-diagram Conventions](./guide/references/instance-diagram-conventions.md)
 
 ### Domain Schemas
```
Comment (aruniverse) [inline docs/bis/leftNav.md:83]: @ColinKerr @diegoalexdiaz can you guys review this section?

File: docs/learning/index.md [hunk 2]
```diff
@@ -1,18 +1,14 @@
-# Documentation Overview
+# Getting started with iTwin.js
 
-This website is organized into sections by tabs at the top of each page.
+iTwin.js has a comprehensive set of APIs that can be used anywhere an [iModel](./imodels.md) may be relevant. It's a big library: where you start will depend on what you'd like to build. If you're looking to extend an iTwin viewer's user interface -- perhaps adding a new tool or widget -- [check out the frontend guide](./frontend/index). For higher level React components, you may want to [take a look at AppUI](./ui/appui/index.md).
```
Comment (aruniverse) [inline docs/learning/index.md:3]: i think we might want to word smith this a bit further

Comment (ben-polinsky) [reply docs/learning/index.md]: I've updated the text for your consideraation.

File: docs/learning/WriteAConnector.md
[the remark this answers is not in the document]
Comment (ben-polinsky) [reply docs/learning/WriteAConnector.md:1]: While I agree it should be reviewed, only grammar and links were updated in this PR. I don't think it should block.

File: docs/bis/leftNav.md (hunk 1 above)
Comment (diegoalexdiaz) [reply docs/bis/leftNav.md:83]: It looks to me that this leftNav is the result of upgrading the one in production according to the newer BeMetalSmith code. There is [another PR](https://github.com/iTwin/itwinjs-core/pull/4893) with several improvements to the BIS docs that also contains changes to this leftNav, including upgrades based on the newer BeMetalSmith.

**Review by diegoalexdiaz — APPROVED** [review]

File: docs/bis/leftNav.md (hunk 1 above)
Comment (ben-polinsky) [reply docs/bis/leftNav.md:83]: Right. Do you want to get that PR through so I can merge it in here?

File: docs/learning/index.md [hunk 3]
```diff
@@ -1,18 +1,12 @@
-# Documentation Overview
+# Getting started with iTwin.js
 
-This website is organized into sections by tabs at the top of each page.
+Welcome to the documentation site for the iTwin.js open-source JavaScript library. iTwin.js offers a versatile and comprehensive set of APIs that can be utilized across various platforms, including web frontends and backends, web services, and desktop, in any context an [iModel](./imodels.md) is used. As a large library, your starting point will depend on the specific development project you have in mind. If you are interested in extending the user interface of an iTwin viewer by adding a new tool or widget, we recommend consulting [the frontend guide](./frontend/index). For higher-level React components, [AppUI](./ui/appui/index.md) may be of interest to you. If you're working on a service or on the server and want to query an iModel's data, you'll want to head over to our [backend documentation](./backend/index). While we're on the subject of querying data, we have a great [tutorial to get you up to speed on ECSQL](./ECSQLTutorial/index.md).
```
Comment (aruniverse) [inline docs/learning/index.md:3]:
```suggestion
Welcome to the documentation site for the iTwin Platform's digital twin open-source JavaScript library, iTwin.js.
```

idk just spit balling here

Comment (aruniverse) [inline docs/learning/index.md:3]:
```suggestion
Welcome to the documentation site for the iTwin Platform's digital twin open-source JavaScript library, iTwin.js. iTwin.js offers a versatile and comprehensive set of APIs that can be utilized across various platforms, including web frontends and backends, web services, and desktop, in any context an [iModel](./imodels.md) is used. As a large library, your starting point will depend on the specific development project you have in mind. If you are interested in extending the user interface of an iTwin viewer by adding a new tool or widget, we recommend consulting [the frontend guide](./frontend/index). For higher-level React components, [AppUI](./ui/appui/index.md) may be of interest to you. If you're working on a service or on the server and want to query an iModel's data, you'll want to head over to our [backend documentation](./backend/index). While we're on the subject of querying data, we have a great [tutorial to get you up to speed on ECSQL](./ECSQLTutorial/index.md).
```

idk just spit balling here

**Review by aruniverse — APPROVED** [review]

File: docs/learning/index.md (hunk 3 above)
Comment (ben-polinsky) [reply docs/learning/index.md:3]: Looks good to me.

File: docs/bis/leftNav.md (hunk 1 above)
Comment (diegoalexdiaz) [reply docs/bis/leftNav.md:83]: The other PR is taking longer to get reviewed than I originally expected, so we better get your PR merged first.

**Comment by ben-polinsky** [comment]
@Mergifyio backport release/4.1.x

**Pull request merged**